### DISCUSSION TITLE
iFrame issue - difference in render in editor mode versus thumbnail view

### DISCUSSION THREAD

**Discussion by jaalibandar** [cables-gl/cables_docs#222] [! Welcome - readme first!]
Attaching two screenshots below : https://cables.gl/p/O1EcMZ

Thumbnail View (Dev)
![image](https://user-images.githubusercontent.com/86233755/162633953-3b72595a-6884-462a-a5e5-b181f9afb23f.png)

Editor Mode Render View (Dev)
![image](https://user-images.githubusercontent.com/86233755/162633976-23d60d8a-ec63-4ca0-b7ed-b958da770fbc.png) 

Shouldn't they be the same? 

What am I trying to do? 
Put an iFrame in the background 

Thumbnail View 
![image](https://user-images.githubusercontent.com/86233755/162634094-73539ee0-de59-4124-a9f7-80cb7058a4bc.png)

Editor Mode Render View 
![image](https://user-images.githubusercontent.com/86233755/162634161-5598a5fc-f280-47ee-b454-33f1e94e8865.png)

I expect both to be same but kindof confused why they aren't.

**Comment by tastez** [comment]
hi @jaalibandar, looks like two issues.

The first one, about overlaying a glcanvas ontop of an iframe - the patch preview page had a forced background for the glcanvas that was interfering - this is now fixed on dev.cables.gl so have a [look](https://dev.cables.gl/p/O1EcMZ). 
You can also force this with CSS yourself in the CSS op inside the patch by putting in    `background-image: none !important;`

Mixing iframes and glcanvases is pretty complicated imo if you aren't into css and browser tricks - but its a pretty cool thing to get into. Z-index, mouse interaction all need some special care and attention and i dont think you can avoid it. Also once we are in that world, exporting your patch to a test website i think is also stress saving because previewing patches that have css and iframes can get funky inside the patch preview page - since it also is part of a website structure with divs and iframes that can overwrite settings - which all don't exist when you are exporting the project or writing your own page to host your patch

ok the second thing is i guess you have a patch but the preview screenshot doesn't look like your patch. Haven't seen exactly what you are doing so I have to assume its because you are using css and iframes in the Urban Treescapes thing? 

CSS, html and Divs aren't part of the glcanvas so a screenshot can't be made automatically inside the editor - but you can do a work around and capture a screenshot in your OS and then upload to yourpatch and generate a screenshot while you put that texture as a fullscreenbackground..then remove the screenshot and save your patch - your patch will have a screenshot you need and match your css/html design.

**Reply by jaalibandar** [reply]
thanks a lot - yeah it is working as expected in dev - the published patch page looks similar to editor view in dev - the iframe background thingy still blacks out in the non dev version - but i expect that will be fixed in some update later. but yeah mixing css with webgl is pretty funky
